File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.py
def solution(n):
    answer = [n]
    
    while True:
        
        if n == 1:
            break
        if n % 2 == 0:
            n /= 2
            answer.append(n)
        elif n % 2 == 1:
            n = 3 * n + 1
            answer.append(n)
    
    return answer

File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int* answer = (int*)malloc(sizeof(int) * 1000);
    int idx = 0;
    
    answer[idx++] = n;
    
    while(n != 1){
        if(n % 2){
            n = 3 * n + 1;
        }else {
            n /= 2;
        }
        answer[idx++] = n;
    }
    
    return answer;
}